src/form/lpac.jl

```diff
@@ -130,3 +130,254 @@ function constraint_ohms_yt_to(pm::AbstractLPACCModel, n::Int, f_bus, t_bus, f_i
+    buses = ref(pm, nw, :bus)
+    branches = ref(pm, nw, :ne_branch)
+
+    va_fr_ne = var(pm, nw)[:va_fr_ne] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :ne_branch)], base_name="$(nw)_va_fr_ne",
+        lower_bound = -2*pi,
+        upper_bound = 2*pi,
+        start = comp_start_value(ref(pm, nw, :bus, branches[i]["f_bus"]), "va_fr_start")
+    )
+
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :va_fr, ids(pm, nw, :ne_branch), va_fr_ne)
+end
+
+""
+function variable_ne_branch_voltage_angle_to(pm::AbstractLPACModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true)
+    buses = ref(pm, nw, :bus)
+    branches = ref(pm, nw, :ne_branch)
+
+    va_to_ne = var(pm, nw)[:va_to_ne] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :ne_branch)], base_name="$(nw)_va_to_ne",
+        lower_bound = -2*pi,
+        upper_bound = 2*pi,
+        start = comp_start_value(ref(pm, nw, :bus, branches[i]["t_bus"]), "va_to_start")
+    )
+
+    # may be bounds can be added in future
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :va_to, ids(pm, nw, :ne_branch), va_to_ne)
+end
+
+
+function variable_ne_branch_cosine(pm::AbstractLPACCModel; nw::Int=pm.cnw, report::Bool=true)
+
+    bi_bp = Dict((i, (b["f_bus"], b["t_bus"])) for (i,b) in ref(pm, nw, :ne_branch))
+    buspair = ref(pm, nw, :ne_buspairs)
+    cs_ne = var(pm, nw)[:cs_ne] = JuMP.@variable(pm.model,
+        [b in ids(pm, nw, :ne_branch)], base_name="$(nw)_cs_ne",
+        lower_bound = 0,
+        upper_bound = 1,
+        start = comp_start_value(ref(pm, nw, :ne_buspairs, bi_bp[b]), "cs_start", 1.0)
+    )
+
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :cs_ne, ids(pm, nw, :ne_branch), cs_ne)
+end
+
+
+function constraint_ne_model_voltage(pm::AbstractLPACCModel, n::Int)
+    buses = ref(pm, n, :bus)
+    branches = ref(pm, n, :ne_branch)
+    bi_bp = Dict((i, (b["f_bus"], b["t_bus"])) for (i,b) in branches)
+    buspair = ref(pm, n, :ne_buspairs)
+    cos_min, cos_max  = ref_calc_angle_difference_bounds(ref(pm, n, :ne_buspairs))
+
+    phi  = var(pm, n, :phi)
+    va  = var(pm, n, :va)
+    z  = var(pm, n, :branch_ne)
+
+    cs = var(pm, n, :cs)
+    cs_ne = var(pm, n, :cs_ne)
+    phi_fr = var(pm, n, :phi_fr_ne)
+    phi_to = var(pm, n, :phi_to_ne)
+    va_fr = var(pm, n, :va_fr_ne)
+    va_to = var(pm, n, :va_to_ne)
+
+
+    for (l,i,j) in ref(pm, n, :ne_arcs_from)
+        JuMP.@constraint(pm.model, phi_fr[l] <= z[l]*(buses[branches[l]["f_bus"]]["vmax"]-1))
+        JuMP.@constraint(pm.model, phi_fr[l] >= z[l]*(buses[branches[l]["f_bus"]]["vmin"]-1))
+
+        JuMP.@constraint(pm.model, cs_ne[l] <= z[l]*cos_max[bi_bp[l]])
+        JuMP.@constraint(pm.model, cs_ne[l] >= z[l]*cos_min[bi_bp[l]])
+
+        JuMP.@constraint(pm.model, phi_to[l] <= z[l]*(buses[branches[l]["t_bus"]]["vmax"]-1))
+        JuMP.@constraint(pm.model, phi_to[l] >= z[l]*(buses[branches[l]["t_bus"]]["vmin"]-1))
```

Comment:
For each of these, there is a simple solution,
```
a_lb, a_ub = _IM.variable_domain(a)
@assert (a_lb <= 0 && a_ub >= 0)
JuMP.@constraint(pm.model, a <= z[l]*a_ub)
JuMP.@constraint(pm.model, a >= z[l]*a_lb)
```
As long as the variables bounds are setup correctly this will work and will be a valid on-off constraint.

Reply:
Slight correction here.  This will only work if the "on" bounds cross zero.  If they do not, then the values of `a_lb, a_ub` should be replaced with the "on" bounds.